job_creator.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+class JobCreator
+{
+    public function getInstallerVersion($githubRepository, $branch)
+    {
+        $repo = explode('/', $githubRepository)[1];
+        if (in_array($repo, NO_INSTALLER_REPOS)) {
+            return '';
+        }
+        $v = explode('.', $branch);
+        if (count($v) == 1) {
+            return '4.x-dev';
+        }
+        if (in_array($repo, LOCKSTEPED_REPOS)) {
+            return '4' . $v[1] . 'x-dev';
+        } else {
+            // use the latest minor version of installer
+            $a = array_keys(INSTALLER_TO_PHP_VERSIONS);
+            $a = array_map(fn($k) => (int) $k, $a);
+            sort($a);
+            return $a[count($a) - 1] . 'x-dev';
```

Comment:
```suggestion
            $installerVersions = array_keys(INSTALLER_TO_PHP_VERSIONS);
            $installerVersions = array_map(fn($k) => (int) $k, $installerVersions);
            sort($installerVersions);
            return $installerVersions[count($installerVersions) - 1] . 'x-dev';
```
I made a guess as to what `$a` is meant to mean here... I don't know what `$k` is at all (I thought maybe the version string but you're casting it to int?), so can you please give that an appropriate name as well?

And maybe add a comment as to what that `array_map` is doing, it's a bit strange. Is it getting the major version number by converting the semver string to int? There's got to be a more readable way of doing that.

Reply:
Updated